pandas/_libs/lib.pyx

```diff
@@ -1284,11 +1287,18 @@ def infer_dtype(value: object, skipna: bool = True) -> str:
     elif hasattr(value, 'dtype'):
         # this will handle ndarray-like
         # e.g. categoricals
+        # begin by checking if there is a tz attribute
+        # in which case we have a Series with dtype DatetimeTZDtype
         try:
             values = getattr(value, '_values', getattr(value, 'values', value))
         except TypeError:
             # This gets hit if we have an EA, since cython expects `values`
             #  to be an ndarray
+            # before using try_infer_map check if there is a tz attribute
```

Comment:
this doesn't belong here at all, this is only for handling a specific dtype w/o special cases, which is what you are doing here.

Reply:
Yes I think I get the idea. I moved the timezone aware Series check above in my new commit.